### PR TITLE
fuse: implement the FUSE_READLINK command

### PR DESCRIPTION
**Pull request by boyuan-he**
Implemented FUSE_READLINK operation

Allow user to send request to read symbolic link on FUSE server.

### CODE REVIEW

File: pkg/abi/linux/fuse.go [hunk 1]
```diff
@@ -401,3 +401,14 @@ func (r *FUSESymLinkIn) MarshalUnsafe(buf []byte) {
 func (r *FUSESymLinkIn) SizeBytes() int {
 	return len(r.Name) + len(r.Target) + 2
 }
+
+// FUSEEmptyIn is used for operations that doesn't have request body.
```
Comment (jinmouil) [inline pkg/abi/linux/fuse.go:405]: "FUSEEmptyIn is used by operations without request body"

"doesn't" is incorrect grammar.

File: pkg/abi/linux/fuse.go [hunk 2]
```diff
@@ -401,3 +401,14 @@ func (r *FUSESymLinkIn) MarshalUnsafe(buf []byte) {
...
+type FUSEEmptyIn struct{}
+
+// MarshalUnsafe do nothing for marshal
```
Comment (jinmouil) [inline pkg/abi/linux/fuse.go:408]: Appends a "."

File: pkg/abi/linux/fuse.go [hunk 3]
```diff
@@ -401,3 +401,14 @@ func (r *FUSESymLinkIn) MarshalUnsafe(buf []byte) {
...
+func (r *FUSEEmptyIn) MarshalUnsafe(buf []byte) {}
+
+// SizeBytes return 0 since on request body.
```
Comment (jinmouil) [inline pkg/abi/linux/fuse.go:411]: "SizeBytes is 0 for empty request."

File: pkg/sentry/fsimpl/fuse/fusefs.go [hunk 4]
```diff
@@ -226,6 +226,9 @@ type Inode struct {
 
 	// imutex serializes changes to a inode.
 	imutex sync.Mutex
+
+	// link is result of following a symbolic link.
```
Comment (jinmouil) [inline pkg/sentry/fsimpl/fuse/fusefs.go:230]: "result" -> "resolved path"?

Better to explicit say it's a cache. If so, please add the "FUSE_CACHE_SYMLINKS" flag to fuse init flags

Comment (jinmouil) [reply pkg/sentry/fsimpl/fuse/fusefs.go:226]: Migrated to #3675

**Comment by mrahatm** [comment]
Merged as part of feature/fuse.

**Pull request closed without merging**